### PR TITLE
Prevent extensibility and enforce properties privacy in foxbox lib. r=azasypkin

### PR DESCRIPTION
**Pull request by gmarty**
@azasypkin: Do you have time for a quick review?

### CODE REVIEW

File: app/js/lib/foxbox/db.js [hunk 1]
```diff
@@ -13,15 +27,18 @@ const DB_TAG_STORE = 'tags';
 
 export default class Db {
   constructor() {
-    this.db = null;
+    // Private properties.
+    this[p.db] = null;
+
+    return Object.seal(this);
```
Comment (azasypkin) [inline app/js/lib/foxbox/db.js]: Honestly I have a mixed feelings about sealing in the constructor, but don't have strong opinion either. Nevertheless I'm open to try :)

P.S This way we "mark" class as non-extendable (maybe that's what we want). In my head, the owner of the instance is the one who should or not seal/freeze the object, but let's see.

Comment (azasypkin) [inline app/js/lib/foxbox/db.js]: __nit:__ I think we don't have to return anything from constructor.

File: app/js/lib/foxbox/foxbox.js [hunk 2]
```diff
@@ -46,11 +47,18 @@ export default class Foxbox extends Service {
   constructor() {
     super();
 
+    // Private properties.
     this[p.settings] = new Settings();
     this[p.db] = new Db();
     this[p.net] = new Network(this[p.settings]);
+    this[p.boxes] = Object.freeze([]);
+    this[p.isPollingEnabled] = null;
```
Comment (azasypkin) [inline app/js/lib/foxbox/foxbox.js]: __nit:__ maybe initialize boolean with ```false```?

File: app/js/lib/foxbox/foxbox.js [hunk 3]
```diff
@@ -46,11 +47,18 @@ export default class Foxbox extends Service {
...
+    this[p.nextPollTimeout] = null;
+
+    // Public properties.
+    this.recipes = null;
 
-    this.boxes = [];
+    return Object.seal(this);
```
Comment (azasypkin) [inline app/js/lib/foxbox/foxbox.js]: __nit:__ same not about redundant ```return``` here

File: app/js/lib/foxbox/network.js [hunk 4]
```diff
@@ -11,18 +12,20 @@ const p = {
...
+    // Private properties.
+    this[p.settings] = settings;
     // Whether we can connect to the box via a local connection.
     this[p.local] = false;
     // Whether we can connect to the box via a remote connection.
     this[p.remote] = false;
     // A reference to the interval to get the online status.
     this[p.pingInterval] = null;
+
+    return Object.seal(this);
```
Comment (azasypkin) [inline app/js/lib/foxbox/network.js]: __nit:__ same not about redundant ```return``` here

File: app/js/lib/foxbox/network.js [hunk 5]
```diff
@@ -65,13 +68,13 @@ export default class Network {
   }
 
   get localOrigin() {
-    return `${this.settings.localScheme}://${this.settings.localHostname}:` +
-      `${this.settings.localPort}`;
+    return `${this[p.settings].localScheme}://` +
```
Comment (azasypkin) [inline app/js/lib/foxbox/network.js]: __optional nit:__ maybe with ```const settings = this[p.settings];``` template string will be less cryptic :)

File: app/js/lib/foxbox/recipes.js [hunk 6]
```diff
@@ -3,16 +3,19 @@
...
-};
+});
 
 export default class Recipe {
   constructor(props) {
+    // Private properties.
     this[p.settings] = props.settings;
     this[p.net] = props.net;
+
+    return Object.seal(this);
```
Comment (azasypkin) [inline app/js/lib/foxbox/recipes.js]: __nit:__ same not about redundant ```return``` here

**Comment by azasypkin** [comment]
Looks good, r+ with nits handled and green Travis (it's not run for some reason)

File: app/js/lib/foxbox/db.js (hunk 1 above)
Comment (gmarty) [inline app/js/lib/foxbox/db.js]: OK, I'm going on with this approach (without the `return`) and we'll re-evaluate later if needed.

**Comment by gmarty** [comment]
Thanks for your review. Nits addressed, so let's see what Travis thinks.

**Pull request merged**